src/main/java/org/folio/service/orders/InvoicesTotalFieldsPopulateService.java

```diff
@@ -0,0 +1,88 @@
+package org.folio.service.orders;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import lombok.extern.log4j.Log4j2;
+import one.util.streamex.StreamEx;
+import org.folio.models.CompositeOrderRetrieveHolder;
+import org.folio.rest.acq.model.invoice.Invoice;
+import org.folio.rest.acq.model.invoice.InvoiceLine;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.service.invoice.InvoiceLineService;
+import org.folio.service.invoice.InvoiceService;
+import org.javamoney.moneta.Money;
+import org.javamoney.moneta.function.MonetaryOperators;
+
+import io.vertx.core.Future;
+
+import static org.folio.orders.utils.HelperUtils.collectResultsOnSuccess;
+
+@Log4j2
+public class InvoicesTotalFieldsPopulateService implements CompositeOrderDynamicDataPopulateService {
+
+  private final InvoiceService invoiceService;
+  private final InvoiceLineService invoiceLineService;
+
+  public InvoicesTotalFieldsPopulateService(InvoiceService invoiceService, InvoiceLineService invoiceLineService) {
+    this.invoiceService = invoiceService;
+    this.invoiceLineService = invoiceLineService;
+  }
+
+  @Override
+  public Future<CompositeOrderRetrieveHolder> populate(CompositeOrderRetrieveHolder holder, RequestContext requestContext) {
+    return Optional.of(holder)
+      .map(CompositeOrderRetrieveHolder::getFiscalYear)
+      .map(v -> withTotalFields(holder, requestContext))
+      .orElseGet(() -> Future.succeededFuture(holder.withTotalExpended(0d).withTotalCredited(0d).withTotalEncumbered(0d)))
+      .onFailure(v -> log.error("Failed at {}.{}", this.getClass().getName(), Thread.currentThread().getStackTrace()[0]));
```

Comment:
better to use v, and better to rename v to e (Exception) or t(Thowable) and in this case Thread.currentThread().getStackTrace() is not needed

Reply:
Good point, this was existing code in TransactionsTotalFieldsPopulateService, will modify